### ISSUE TITLE
MD050: false-positive when handling __X__ in mkdocs-string sections

### ISSUE THREAD

**Issue by noamgot** [rvben/rumdl#515]
### rumdl version

0.1.145

### Installation method

Other

### Bug description

(related to #17 )

I still see an issue that I consider as a false-positive when using mkdocs-strings (rumdl==0.1.145):

```md
# Example

We have here some **bold text**.

::: my_module.MyClass
    options:
      members:
        - __init__

```

```
❯ rumdl check --flavor mkdocs example.md --fix
example.md:8:11: [MD050] Strong emphasis should use ** instead of __ [fixed]

Fixed: Fixed 1/1 issues in 1 file (16ms)
```
For this particular case, I expect rumdl to ignore the `__init__` and leave it as is; I'm not sure if it should be default, or configurable by the user (e.g., add some `ignore` attribute to the rule that accept patterns to ignroe). 

### Steps to reproduce

described above

### Expected behavior

`__X__` methods in mkdocs-string blocks should be ignored (I think?)

### Actual behavior

`__X__` is converted to `**X**`

### Configuration

```toml
# rumdl configuration file

# Inherit settings from another config file (relative to this file's directory)
# extends = "../base.rumdl.toml"

# Global configuration options
[global]
# List of rules to disable (uncomment and modify as needed)
# disable = ["MD013", "MD033"]

# List of rules to enable exclusively (replaces defaults; only these rules will run)
# enable = ["MD001", "MD003", "MD004"]

# Additional rules to enable on top of defaults (additive, does not replace)
# Use this to activate opt-in rules like MD060, MD063, MD072, MD073, MD074
# extend-enable = ["MD060", "MD063"]

# Additional rules to disable on top of the disable list (additive)
# extend-disable = ["MD041"]

# List of file/directory patterns to include for linting (if provided, only these will be linted)
# include = [
#    "docs/*.md",
#    "src/**/*.md",
#    "README.md"
# ]

# List of file/directory patterns to exclude from linting
exclude = [
    # Common directories to exclude
    ".git",
    ".github",
    "node_modules",
    "vendor",
    "dist",
    "build",
    ".pixi",
    ".venv",

    # Specific files or patterns
    "CHANGELOG.md",
    "LICENSE.md",
]

# Respect .gitignore files when scanning directories (default: true)
respect-gitignore = true

# Markdown flavor/dialect (uncomment to enable)
# Options: standard (default), gfm, commonmark, mkdocs, mdx, quarto

[per-file-flavor]
"docs/**/*.md" = "mkdocs"

# Rule-specific configurations (uncomment and modify as needed)

# [MD003]
# style = "atx"  # Heading style (atx, atx_closed, setext)

# [MD004]

# [MD007]
# indent = 4  # Unordered list indentation

[MD013]
line-length = 120  # Line length
reflow = true
code-blocks = false  # Exclude code blocks from line length check
# tables = false  # Exclude tables from line length check
# headings = true  # Include headings in line length check

# [MD044]
# names = ["rumdl", "Markdown", "GitHub"]  # Proper names that should be capitalized correctly
# code-blocks = false  # Check code blocks for proper names (default: false, skips code blocks)
```

### Operating system

Linux

**Comment by rvben** [comment]
Fixed in v0.1.46. Emphasis-related rules (MD049/MD050) now skip mkdocstrings blocks, so `__init__` inside `::: module` sections is no longer flagged. Thanks for reporting @noamgot!
